### DISCUSSION TITLE
How to perform custom animation development？

### DISCUSSION THREAD

**Discussion by Grieey** [slint-ui/slint#2305] [Q&A] [answered]
How to implement animations within a build-in widget? For example, I want to create an animation for the cursor movement in a TextInput.

**Comment by tronical** [comment, marked as answer]
Hi! I’m afraid that the text cursor cannot be animated. This would have to be implemented in Slint directly.

**Reply by bjorn** [reply]
I think for such a use-case, it may suffice if the cursor's position (or rectangle) would be exposed and the cursor could be hidden. Then, any of the existing primitives can be used to implement a different appearance along with animation.

**Reply by tronical** [reply]
Excellent idea

**Reply by ogoffart** [reply]
There is already a callback `cursor-position-changed` in TextInput  https://slint-ui.com/snapshots/master/docs/slint/src/builtins/elements.html#id19

What's missing is a way to hide the cursor.  Maybe setting `text-cursor-width` would work.
